contracts/SnappAuctionChallenge.sol

```diff
@@ -366,25 +356,25 @@ contract SnappAuctionChallenge {
         uint offset,
         uint length,
         uint width
-    ) internal view returns (bytes32) {
+    ) internal pure returns (bytes32) {
         bytes16 order = 0x0;
-        bytes32 rollingHash = startingHash;
+        bytes32 rolledHash = startingHash;
         for (uint256 i = 32 + offset; i <= 32 + offset + length; i += width) {
             /* solhint-disable no-inline-assembly */
             assembly {
                 order := mload(add(data, i))
             }
-            rollingHash = sha256(abi.encodePacked(rollingHash, order));
+            rolledHash = sha256(abi.encodePacked(rolledHash, order));
         }
-        return rollingHash;
+        return rolledHash;
     }
 
-    function challengePriceNonUniform(
-        uint buyPrice, 
-        uint sellPrice, 
-        uint buyVolume, 
+    function priceNotUniform(
+        uint buyPrice,
+        uint sellPrice,
+        uint buyVolume,
         uint sellVolume
     ) internal pure returns (bool) {
-        return ((buyPrice * sellVolume) - (sellPrice * buyVolume))**2 > EPSILON**2;
+        return ((buyPrice * buyVolume) - (sellPrice * sellVolume))**2 > EPSILON**2;
```

Comment:
@fleupold - I wanted to point out here that we had to switch thew buy and sell volumes in this calculation!

Furthermore, I think this might also be a good place for a safe-math computation or even writing our own (safe) version of absolute value... since we would have an overflow in our difference when
```
(buyPrice * buyVolume) < (sellPrice * sellVolume)
```

which could result in a false positive!

Reply:
Isn't it `buyPrice/sellPrice = buyVolume/sellVolume`?

Good point about the overflow logic. I'd hope we could just use SafeMath here though.

Reply:
Nah, that ratio you are thinking of is for limit price (not uniform clearing). If we were to you the term amount here (which is technically more accurate), then `volume = price * amount`. To see that we have a uniform clearing price we check that `buyVolume = sellVolume` (i.e. when compared with each other without regards to a currency). That is to say, we want 

```
buyPrice * buyAmount = buyVolume = sellVolume = sellPrice * sellAmount
```

Also, as for using safe math, I do not think this is a situation where safe math would help. Because of rounding, we do not know which of `buyVolume` or `sellVolume` will actually be greater. Safe math reverts with an error when overflow occurs. What we really want here is a proper absolute value

```
abs(x) = -x if x < 0 else x
```
I can try and put something together in solidity, but it would have to be specific to differences. 

```js
function absDiff(a, b) returns uint {
    if a > b {
        return a - b;
    } else {
        return b - a;
}
```

Anyway, if we don't have this, I am afraid this will be a very undesired bug.

Reply:
I have pushed a change that should cover all the bases here! Check it out. Also we might save some gas by not having to square the terms... Not sure how much really.

Reply:
I see your point now, thanks for clarifying. I also agree `absDiff` seems concise enough and should be used instead of safeMath here.

> Also we might save some gas by not having to square the terms... Not sure how much really.

Would you be interested in finding out? You could write a simple test case and use the gas reporter to measure it.

Reply:
> Would you be interested in finding out? 

I would be interested in finding out, but I am not sure of the best approach. Should I run the test once with the old code and once with the new, or should I just update the gas estimate on the design-issue with whatever I get using the new code? 

Should this be a standalone test only checking whether the if branch uses more gas than squaring a difference? 

Could we maybe do a small pair-coding session where you show how your estimates were obtained in the first place?